app/controllers/api_controller/authentication.rb

```diff
@@ -44,5 +49,14 @@ def require_api_user_or_token
     def userid_to_userobj(userid)
       User.find_by_userid(userid)
     end
+
+    def fetch_and_validate_requester_type
+      requester_type = params['requester_type']
+      if requester_type && !REQUESTER_TYPES.include?(requester_type)
+        requester_types = REQUESTER_TYPES.join(', ')
+        raise BadRequestError, "Invalid requester_type #{requester_type} specified, valid types are: #{requester_types}"
+      end
+      requester_type
```

Comment:
np, can use fetch, minor update to above though fetch returns value, need to return key.